### PR TITLE
Layer system update: asynchronous loading, renames, new configuration

### PR DESCRIPTION
**Pull request by kb173**

### CODE REVIEW

**Comment by kb173** [comment]
When merged this closes #303 and #290

File: UI/Layers/LayerConfiguration/ItemList.gd [hunk 1]
```diff
@@ -0,0 +1,35 @@
+extends ItemList
+
```
Comment (MathiasBaumgartinger) [inline UI/Layers/LayerConfiguration/ItemList.gd:2]: Add short explanation what this does:

```
Display of all geolayers and handling for z-index of geo-layers in the 2D space.
```

Comment (MathiasBaumgartinger) [reply UI/Layers/LayerConfiguration/ItemList.gd:2]: Also rename this from `ItemList.gd` to `GeoLayersList.gd`

File: UI/Layers/LayerConfiguration/LayerCompositionWidget.gd [hunk 2]
```diff
@@ -25,34 +25,53 @@ func _ready():
...
-			icon.texture = load("%s/map.svg" % icon_prefix)
-		LayerComposition.RenderType.VEGETATION:
-			icon.texture = load("%s/grass.svg" % icon_prefix)
-		_:
-			icon.texture = load("%s/layer.svg" % icon_prefix)
+	if layer_composition.render_info is LayerComposition.ObjectRenderInfo:
+		icon.texture = load("%s/vector.svg" % icon_prefix)
+	else:
+		icon.texture = load("%s/raster.svg" % icon_prefix)
+	# FIXME: Reimplement with render_info
```
Comment (MathiasBaumgartinger) [inline UI/Layers/LayerConfiguration/LayerCompositionWidget.gd:42]: What to do with that? Let's not leave those things in a FIXME state for 2 years :)

File: UI/Layers/LayerConfiguration/Misc/LayerCompositionConfigurationWindow.gd [hunk 3]
```diff
@@ -74,16 +71,10 @@ func _on_confirm():
...
-#func resize(add: Vector2):
-#	minimum_size.y = min_size.y + add.y
-#	minimum_size.x = max(add.x, min_size.x) 
-#	size = minimum_size
-
 
 func _add_types():
 	var idx = 0
-	for type in LayerComposition.RenderType:
+	for type in LayerComposition.RENDER_INFOS.keys():
```
Comment (MathiasBaumgartinger) [inline UI/Layers/LayerConfiguration/Misc/LayerCompositionConfigurationWindow.gd:77]: Does this work? I feel like committing (i.e. pressing ok) might cause an error when loading a new layer

File: UI/Layers/LayerManagement/AbstractLayerManagement.gd [hunk 4]
```diff
@@ -4,11 +4,11 @@ class_name AbstractLayerManagement
```
Comment (MathiasBaumgartinger) [inline UI/Layers/LayerManagement/AbstractLayerManagement.gd:4]: Maybe add a `FIXME: move this from layercompositions to geolayers`

**Review by MathiasBaumgartinger — CHANGES_REQUESTED** [review]

**Pull request merged**